tests/tests_insights_ansible_host.yaml

```diff
@@ -0,0 +1,71 @@
+# SPDX-License-Identifier: MIT
+---
+- name: Insights ansible host test
+  hosts: all
+  gather_facts: false
+  become: true
+
+  tasks:
+    - name: Setup Insights
+      import_tasks: tasks/setup_insights.yml
+
+    - name: Test ansible_host
+      block:
+        - name: Add ansible_host and register insights
+          include_role:
+            name: linux-system-roles.rhc
+            public: true
+          vars:
+            rhc_auth:
+              login:
+                username: "{{ lsr_rhc_test_data.reg_username }}"
+                password: "{{ lsr_rhc_test_data.reg_password }}"
+            rhc_insights:
+              ansible_host: test-host
+              remediation: absent
+              state: present
+            rhc_organization: "{{ lsr_rhc_test_data.reg_organization }}"
+            rhc_server:
+              hostname: "{{ lsr_rhc_test_data.candlepin_host }}"
+              port: "{{ lsr_rhc_test_data.candlepin_port }}"
+              prefix: "{{ lsr_rhc_test_data.candlepin_prefix }}"
+              insecure: "{{ lsr_rhc_test_data.candlepin_insecure }}"
+            rhc_baseurl: "{{ lsr_rhc_test_data.baseurl | d(omit) }}"
+
+        - name: Check ansible host is set to 'test-host' in config file
+          command:
+            grep -ixq "^ansible_host=test-host" {{ __rhc_insights_conf }}
+          changed_when: true
+
+        - name: Change ansible host to 'new-test-host'
+          include_role:
+            name: linux-system-roles.rhc
+          vars:
+            rhc_insights:
+              ansible_host: new-test-host
+              remediation: absent
+
+        - name: Check ansible_host is set to 'new-test-host' in config file
+          command:
+            grep -ixq "^ansible_host=new-test-host" {{ __rhc_insights_conf }}
+          changed_when: true
+
+        - name: Change ansible host to a null value (noop)
+          include_role:
+            name: linux-system-roles.rhc
+          vars:
+            rhc_insights:
+              ansible_host: null
+              remediation: absent
+
+        - name: Check ansible_host has not changed in config file
+          command:
+            grep -ixq "^ansible_host=new-test-host" {{ __rhc_insights_conf }}
+          changed_when: true
+
```

Comment:
I think it could be a good idea to add a test for removing the display-host name, and another to check if after unregistration the display-name also change.

Reply:
yes, see the unregistration task later on in the `always:` block